src/uwtools/tests/cli/test_run_forecast.py

```diff
@@ -39,7 +41,21 @@ def test_main(files):
     with patch.object(run_forecast, "parse_args", return_value=args):
         with patch.object(run_forecast.forecast, "FV3Forecast") as fv3fcst:
             run_forecast.main()
-            fv3fcst.assert_called_once_with(cfgfile, machinefile, log_name="run-forecast")
+            fv3fcst.assert_called_once_with(
```

Comment:
The current main bug seems to be the syntax here is not actually successfully calling `FV3Forecast`. I've tried rewriting it more like the `test_main()` in `test_set_config()` but with no success. I believe `run_forecast.main()` is set up ok, so any observations on this are appreciated

Reply:
Try `assert fv3fcst.__init__.called_once_with(...)`? It's the constructor of the class that gets called when `FV3Forecast` (or a mock of it) gets instantiated.

Reply:
Interesting - I think I might still be misunderstanding the way python handles the constructor vs the class, thats a good one!

..Ah, the type checker rapped my knuckles on that, "error: Accessing "__init__" on an instance is unsound"

Reply:
If this simulation is realistic, what you had first looks like it should work:
```
% cat foo.py
class C:
    pass
```
Then in the Python REPL:
```
>>> from unittest.mock import patch
>>> import foo
>>> with patch.object(foo, "C") as C:
...     C(a=1, b=2)
...     C.assert_called_once_with(a=1, b=2)
... 
<MagicMock name='C()' id='281472952832592'>
```
I didn't need to say `C.__init__` was called, just `C`. I'm going to check out your branch and see if I can reproduce whatever error you're getting.

Reply:
I think I see what's going on. In `uwtools.cli.run_forecast`, you have
```
    forecast_class = getattr(forecast, f"{args.forecast_model}Forecast")
    try:
        forecast_class.run(
            config_file=args.config_file,
            dry_run=args.dry_run,
            forecast_app=args.forecast_app,
            forecast_model=args.forecast_model,
            log_file=args.log_file,
            machine=args.machine,
            quiet=args.quiet,
            verbose=args.verbose,
        )
```
The line `forecast_class = getattr(forecast, f"{args.forecast_model}Forecast")` gives you a reference to the `FV3Forecast` class, on which you then make the call `forecast_class.run(...)`. This would work if `run()` were a static method. But it's an instance method, so you first need to instantiate an `FV3Forecast` object and _then_ call the `run()` method on that instance, like
```
    forecast_class = getattr(forecast, f"{args.forecast_model}Forecast")
    forecast_obj = forecast_class()
    try:
        forecast_obj.run(...)
```
This would have been apparent if you'd tried running `run_forecast.main()` outside of a unit test with mocking going on, because Python would have complained about `run()` not being something you can call on the `FV3Forecast` **class**. (And `pylint` and probably `mypy` would have complained if we weren't using `getattr()` to obtain a reference to the `FV3Forecast` class -- if we literally had `FV3Forecast` in `run_forecast.main()`, the static-analysis tools could have done their job, but the dynamic `getattr()` lookup hides that from them.)

But mocking makes the impossible possible, in a rare, unhelpful way in this case. I was already writing up what follows to share, but it just became especially pertinent, so here goes:

--

I should probably share something that I lost a lot of time by not understanding when I started working with `unittest.mock`: When you have a `Mock` object and reference a member, either as a property (`mymock.property`) or a method (`mymock.method()`), `Mock` _creates those members_ dynamically as you reference them, and to any depth of dot-chaining. They just spring into existence, e.g.
```
>>> from unittest.mock import Mock
>>> m = Mock()
>>> type(m.property)
<class 'unittest.mock.Mock'>
>>> type(m.n.o.func)
<class 'unittest.mock.Mock'>
>>> m.n.o.func(foo=88, bar=99)
<Mock name='mock.n.o.func()' id='281472987072064'>
>>> m.n.o.func.assert_called_once_with(foo=88, bar=99)
>>> 
```
I didn't have to do anything special to create `n` as a member of `m`, `o` as a member of `n`, etc.

--

So, this is why `forecast_class.run()` doesn't crash via the unit test: `forecast_class` is a `Mock` (probably `MagicMock`, but the difference isn't important here) object, so a `.run` member is automatically created upon reference. `Mock` objects are callable, so it's fine that that member is called as a function/method. But the _constructor_ of `FV3Forecast` is never called because we never instantiate an object.

Upon reflection, the line `forecast_class.run(` should have told us everything: An `FV3Forecast` **class** doesn't have a `run()` (static) method to call; but a `FV3Forecast` **instance** has a `run()` (instance) method to call.

Reply:
Hooray, for lack of an insightful lightbulb emoji! `Mock` is still new to me and I'm a little clumsy with constructors, so that was what I had a feeling the issue was, but its a more fundamental 'gotcha' that is hard to parse and practically impossible for usual patchwork solutions to fix. Thanks, I'm going to have to really reread this last part to internalize it!

Reply:
Assuming what I wrote is correct, I found this out by inserting a `breakpoint()` in the right place, dropping into the Python debugger, and stepping through the code, introspecting variables as I went. Maybe a quick tutorial on `pdb` would be useful for our team. It's a really powerful (and surprisingly simple) way to delve into weird problems.

Reply:
Yep, going to make sure to use breakpoints now, that'll help a lot. I had gotten tied up blaming the mock patching for the failure to make the instance right when it was just the darn `getattr()` again. It worked correctly this way, although there was some fussing in passing the args correctly. However, with moving all of those args out it should lessen the headache by far